### DISCUSSION TITLE
How to develop a game without utiLITI (instanciate an environment programmatically)?

### DISCUSSION THREAD

**Discussion by HarleyDavidson86** [gurkenlabs/litiengine#823] [General]
On my way to look for a java based game engine I stumbled upon LITIENGINE, which looks very interesting to me.
To get in touch with a new framework I always like to start off with something really simple like a clone of "snake" or "SpaceImpact" known from nokia mobilephones.

In the FAQ there is a section telling me that I can create games without using the editor.
So how can I create an environment by myself, because as I can see I need to instanciate one to add entities like enemies or the player?

**Comment by nightm4re94** [comment]
If you want to instantiate an `Environment` programmatically, you'll have to understand that there is no `Environment` without an `IMap`. So even if you want to have a blank canvas with a black background, you'll have to create an `IMap` that contains information about the playing field.

Create an entire `IMap` by code, adding `MapObjects` and `Tiles` procedurally using the [`MapGenerator`](https://github.com/gurkenlabs/litiengine/blob/main/litiengine/src/main/java/de/gurkenlabs/litiengine/resources/Maps.java#L90). Standard objects like CollisionBoxes, Props, etc. can already be part of the Map as `MapObjects` so the corresponding `IEntity` type will be loaded when loading the map. This would also allow you to export the generated map and open it with utiLITI . Then you would call `Game.world().loadEnvironment(IMap map)` to instantiate an `Environment` from the map you've created. If you are unfamiliar with the `MapObject` specifications and how to create an `IMapObject` that will be loaded as a Prop, for example, you can keep the information in the `IMap` really minimal and make modifications to the environment's `IEntity`s after loading it. The downside is that you won't have any persistency for the added `IEntity`s and you couldn't export your map like if you had declared `MapObject`s on the `IMap` itself.
